db/src/urls.rs

```diff
@@ -52,3 +62,48 @@ pub async fn add_url(
         ))
     }
 }
+
+pub async fn fetch_url_from_pages_table(pool: &PgPool, page_id: Uuid) -> color_eyre::Result<Page> {
+    let result = sqlx::query_as!(
+        Page,
+        "SELECT page_id, user_id, url FROM Pages WHERE page_id = $1",
+        page_id
+    )
+    .fetch_one(pool)
+    .await?;
+
+    Ok(result)
+}
+
+async fn download_raw_html(_pool: &PgPool, url: &str) -> color_eyre::Result<String> {
+    let response = reqwest::get(url).await.map_err(color_eyre::Report::from)?;
+    let html = response.text().await.map_err(color_eyre::Report::from)?;
+    Ok(html)
+}
+
+async fn store_raw_html_in_page_snapshot(
+    pool: &PgPool,
+    page: Page,
+) -> color_eyre::Result<PageSnapShot> {
+    let raw_html = download_raw_html(pool, &page.url).await?;
+    let current_time = chrono::Utc::now();
+
+    let result = sqlx::query_as!(
+        PageSnapShot,
+        "INSERT INTO PageSnapShot (raw_html, fetched_at, page_id) 
+         VALUES ($1, $2, $3) RETURNING page_snapshot_id, page_id, raw_html, fetched_at",
```

Comment:
If you change this to a `RETURNING *` it will fail to compile if we add a new column and do NOT add it to the `PageSnapShot` struct here

Which _can_ be nice if we want that struct to always be the 'full' DB row. No strong feels here just chatting!

Reply:
Make sense!